File: back/src/main/java/fr/semifir/localib/models/Location.java
package fr.semifir.localib.models;

import lombok.Data;

import javax.persistence.*;
import java.util.Date;

@Entity
@Data
public class Location {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(name = "date_debut")
    private Date dateDebut;
    @Column(name = "date_fin")
    private Date dateFin;
    @Column(name = "prix_total")
    private double prixTotal;

    @ManyToOne
    @JoinColumn(name = "vehicule_id")
    private Vehicule vehicule;

    @ManyToOne
    @JoinColumn(name = "locataire_id")
    private Locataire locataire;

}

File: front/src/app/models/location.ts
import { Locataire } from "./locataire";
import { Vehicule } from "./vehicule";

export class Location {

    constructor(
        public id : string,
        public dateDebut : Date,
        public dateFin : Date,
        public prixTotal : number,
        public locataire : Locataire,
        public vehicule : Vehicule
    ) {

    }
}